### PR TITLE
Fix check gpu index logic in local mode

### PR DESCRIPTION
**Pull request by SparkSnail**
In local training service, NNI will check if the gpu index is valid. But there is a kind of scene that gpu_metric file is not generated yet, the gpuCount number equals to 0, but check gpu index will cause experiment failed.

### CODE REVIEW

File: ts/nni_manager/training_service/local/localTrainingService.ts [hunk 1]
```diff
@@ -436,7 +436,7 @@ class LocalTrainingService implements TrainingService {
 
     private checkSpecifiedGpuIndices(): void {
         const gpuCount: number = this.gpuScheduler.getSystemGpuCount();
-        if (this.designatedGpuIndices !== undefined) {
+        if (this.designatedGpuIndices !== undefined && gpuCount > 0) {
```
Comment (liuzhe-lz) [inline ts/nni_manager/training_service/local/localTrainingService.ts:439]: What will happen if gpu count IS 0?

Comment (SparkSnail) [reply ts/nni_manager/training_service/local/localTrainingService.ts:439]: Only detect gpu count when user set gpuNum>0 in config file. If gpu count is 0 in the machine, the trial job will always be waitting. We could not guarantee the detected 0 is a valid number in gpuScheduler component, then should not validate gpuIndices when count is 0.

Comment (liuzhe-lz) [reply ts/nni_manager/training_service/local/localTrainingService.ts:439]: I suggest to return undefined or null before first detection. `getSystemGpuCount()` is only used here so I guess it won't be too much effort.

Comment (SparkSnail) [reply ts/nni_manager/training_service/local/localTrainingService.ts]: agree, updated.

**Review by liuzhe-lz — APPROVED** [review]

**Review by J-shang — APPROVED** [review]

**Pull request merged**